### PR TITLE
Issue 17772 binary validations

### PR DESCRIPTION
**Pull request by jgambarios**
With this change the properties allowedFileTypes and maxFileLength when added to a Binary field into a content type are being respects.

maxFileLength supports conventions such as 1kb, 1mb, 1gb
allowedFileTypes supports mime type with wildcards such as text/*, application/*, or specific ones such as application/pdf, separated by commas.

So, When checkin a contentlet with a binary fields, if these validations are in placed, will be fired and throw an exception if not compliance.

In addition, a small change has been made in order to support hooks by osgi when called the validateContentlet inside the checkin.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.dotcms.rest.api.v1.temp;
 
+import com.dotcms.util.ConversionUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.util.ConversionUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        final int stringWidth             = fontMetrics.stringWidth(message);
+        final int stringHeight            = fontMetrics.getAscent();
+        graphics2D.setPaint(Color.black);
+        graphics2D.drawString(message, (width - stringWidth) / 2, height / 2 + stringHeight / 4);
+
+        ImageIO.write(bufferedImage, "PNG", tempTestFile);
+
+        final String mimeType = MimeTypeUtils.getMimeType(tempTestFile);
+
+        Assert.assertEquals("The mime type should be image/png","image/png", mimeType);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java [hunk 3]
```diff
@@ -13,6 +13,40 @@
...
+        Assert.assertEquals(1024l, value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("112kb", -1l);
+        Assert.assertEquals(112l * 1024l, value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("3kb", -1l);
+        Assert.assertEquals( (3l * 1024l), value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("2mb", -1l);
+        Assert.assertEquals( (2l * 1024l * 1024l), value);
```
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
+package com.dotcms.util;
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business.RoleAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java [hunk 6]
```diff
@@ -13,6 +13,40 @@
...
+        Assert.assertEquals(9000l, value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("1kb", -1l);
+        Assert.assertEquals(1024l, value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("112kb", -1l);
+        Assert.assertEquals(112l * 1024l, value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("3kb", -1l);
+        Assert.assertEquals( (3l * 1024l), value);
```
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.portlets.workflows.model.WorkflowState;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.util.StringPool;
+import com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageWriter;
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'io.vavr.Tuple2'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
+package com.dotcms.util;
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.contenttype.business.ContentTypeAPIImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+        final File tempTestFile = File
+                .createTempFile("image" + new Date().getTime(), ".png");
+        final int width         = 200;
+        final int height        = 200;
+        final BufferedImage bufferedImage = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+        final Graphics2D   graphics2D     = bufferedImage.createGraphics();
+        final Font font                   = new Font("TimesRoman", Font.BOLD, 20);
+        final String message              = "Test";
+        FontMetrics fontMetrics           = graphics2D.getFontMetrics();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Local variable 'fontMetrics' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.portlets.workflows.model.WorkflowState;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.model.WorkflowState'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.dotcms.util.CollectionsUtils.*;
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+    @Test
+    public void test_getMimeType_text_plain() throws IOException {
+
+        final File tempTestFile = File
+                .createTempFile("csvTest_" + new Date().getTime(), ".txt");
+        FileUtils.writeStringToFile(tempTestFile, "Test");
+
+        final String mimeType = MimeTypeUtils.getMimeType(tempTestFile);
+
+        Assert.assertEquals("The mime type should be application/text","text/plain", mimeType);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
+package com.dotcms.util;
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.TestDataUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.portlets.workflows.model.WorkflowState;
+import com.dotmarketing.util.UUIDGenerator;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.util.UUIDGenerator'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+import static org.junit.Assert.*;
+
+/**
+ * MimeTypeUtils unit test.
+ * @author jsanca
+ */
+public class MimeTypeUtilsTest extends UnitTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.portlets.workflows.model.WorkflowState;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.util.StringPool;
+import com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageWriter;
+import io.vavr.Tuple;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'io.vavr.Tuple'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 20]
```diff
@@ -0,0 +1,86 @@
+package com.dotcms.util;
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.TestUserUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
+import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
+import com.dotmarketing.portlets.workflows.model.WorkflowState;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.util.StringPool;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.portlets.workflows.actionlet.PublishContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.SaveContentAsDraftActionlet;
+import com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.workflows.actionlet.UnpublishContentActionlet'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+        String mimeType = Sneaky.sneak(() -> Files.probeContentType(path));
+
+        if  (!UtilMethods.isSet(mimeType)) {
+
+            mimeType    = Config.CONTEXT.getMimeType(binary.getAbsolutePath());
+
+            if( !UtilMethods.isSet(mimeType)){
+                try {
+                    mimeType = new TikaUtils().detect(binary);
+                } catch(Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 24]
```diff
@@ -0,0 +1,86 @@
+package com.dotcms.util;
+
+import com.dotcms.UnitTestBase;
+import com.dotcms.contenttype.business.ContentTypeAPIImpl;
+import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.repackage.org.apache.commons.io.FileUtils;
+import com.dotmarketing.business.APILocator;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.business.APILocator'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 25]
```diff
@@ -0,0 +1,86 @@
...
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.dotcms.util.CollectionsUtils.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.util.CollectionsUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Date;
+
+/**
+ * MimeTypeUtils unit test.
+ * @author jsanca
+ */
+public class MimeTypeUtilsTest extends UnitTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 27]
```diff
@@ -33,10 +10,25 @@
...
-
 import io.vavr.control.Try;
+import org.glassfish.jersey.media.multipart.BodyPart;
+import org.glassfish.jersey.media.multipart.ContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.server.JSONP;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.*;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java (hunk 23 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java [hunk 28]
```diff
@@ -13,6 +13,40 @@
...
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("2mb", -1l);
+        Assert.assertEquals( (2l * 1024l * 1024l), value);
+
+        value = ConversionUtils.toLongFromByteCountHumanDisplaySize("4gb", -1l);
+        Assert.assertEquals( (4l * 1024l * 1024l * 1024l), value);
```
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Mime Type Utils
+ * @author jsanca
+ */
+public class MimeTypeUtils {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java]: Codacy Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

**Review by dsolistorres — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java (hunk 28 above)
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java (hunk 27 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java (hunk 26 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/MimeTypeUtilsTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java (hunk 23 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java (hunk 29 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/MimeTypeUtils.java]: Codacy Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

File: dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotcms/util/ConversionUtilsTest.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4801734)

**Pull request merged**